easybuild/tools/filetools.py

```diff
@@ -760,6 +760,17 @@ def verify_checksum(path, checksums):
         checksums = [checksums]
 
     for checksum in checksums:
+        if isinstance(checksum, dict):
+            filename = os.path.split(path)[-1]
```

Comment:
You should use `os.path.basename` here:

```python
filename = os.path.basename(path)
```

Or even better, do this way on top of this function so you can also use `filename` above.